pkg/daemon/discover/discover.go

```diff
@@ -372,9 +403,51 @@ func probeDevices(context *clusterd.Context) ([]sys.LocalDisk, error) {
 		device.Filesystem = fs
 		device.Empty = clusterd.GetDeviceEmpty(device)
 
+		// Add the information provided by ceph-volume inventory
+		if cvInventory != nil {
+			CVData, device_exists := (*cvInventory)[path.Join("/dev/", device.Name)]
+			if device_exists {
+				device.CephVolumeData = CVData
+			} else {
+				logger.Errorf("ceph-volume information for device %s not found", device.Name)
+			}
+		} else {
+			device.CephVolumeData = ""
+		}
+
 		devices = append(devices, *device)
 	}
 
 	logger.Infof("available devices: %+v", devices)
 	return devices, nil
 }
+
+func GetCephVolumeInventory(context *clusterd.Context) (*map[string]string, error) {
+	inventory, err := context.Executor.ExecuteCommandWithOutput(false, "", "ceph-volume", "inventory", "--format", "json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute ceph-volume inventory: %+v", err)
+	}
+
+	//Return a map with the information of each device indexed by path
+	CVDevices := make(map[string]string)
+
+	// Get a slice to store the json data
+	bInventory := []byte(inventory)
+	var CVInventory []CephVolumeInventory
+	err = json.Unmarshal(bInventory, &CVInventory)
+	if err != nil {
+		logger.Errorf("error unmarshaling json data coming from ceph-volume inventory: %v", err)
+		return &CVDevices, nil
+	}
+
+	for _, device := range CVInventory {
+		jsonData, err := json.Marshal(device)
+		if err != nil {
+			logger.Errorf("error marshaling json data for device: %v", device.Path)
```

Comment:
return error

Reply:
In this case, to have an error marshaling one device json does not imply that others devices have the same problem.
In this way we have a result with the parseable devices, if i return the error, in the better case we will only have the parsed devices until the first one with error.